server/requirements.txt

```diff
@@ -17,63 +17,75 @@ filelock==3.15.4 ; python_version >= "3.9" and python_version < "3.13"
 fsspec==2024.3.1 ; python_version >= "3.9" and python_version < "3.13"
 fsspec[http]==2024.3.1 ; python_version >= "3.9" and python_version < "3.13"
-googleapis-common-protos==1.63.1 ; python_version >= "3.9" and python_version < "3.13"
+googleapis-common-protos==1.63.2 ; python_version >= "3.9" and python_version < "3.13"
 grpc-interceptor==0.15.4 ; python_version >= "3.9" and python_version < "3.13"
 grpcio-reflection==1.48.2 ; python_version >= "3.9" and python_version < "3.13"
 grpcio-status==1.48.2 ; python_version >= "3.9" and python_version < "3.13"
-grpcio==1.64.1 ; python_version >= "3.9" and python_version < "3.13"
-hf-transfer==0.1.6 ; python_version >= "3.9" and python_version < "3.13"
+grpcio==1.65.4 ; python_version >= "3.9" and python_version < "3.13"
+hf-transfer==0.1.8 ; python_version >= "3.9" and python_version < "3.13"
 huggingface-hub==0.22.2 ; python_version >= "3.9" and python_version < "3.13"
 humanfriendly==10.0 ; python_version >= "3.9" and python_version < "3.13"
 idna==3.7 ; python_version >= "3.9" and python_version < "3.13"
-importlib-metadata==7.2.1 ; python_version >= "3.9" and python_version < "3.13"
-intel-openmp==2021.4.0 ; python_version >= "3.9" and python_version < "3.13" and platform_system == "Windows"
+importlib-metadata==8.0.0 ; python_version >= "3.9" and python_version < "3.13"
 jinja2==3.1.4 ; python_version >= "3.9" and python_version < "3.13"
 loguru==0.6.0 ; python_version >= "3.9" and python_version < "3.13"
 markupsafe==2.1.5 ; python_version >= "3.9" and python_version < "3.13"
-mkl==2021.4.0 ; python_version >= "3.9" and python_version < "3.13" and platform_system == "Windows"
 mpmath==1.3.0 ; python_version >= "3.9" and python_version < "3.13"
 multidict==6.0.5 ; python_version >= "3.9" and python_version < "3.13"
 multiprocess==0.70.16 ; python_version >= "3.9" and python_version < "3.13"
 networkx==3.2.1 ; python_version >= "3.9" and python_version < "3.13"
 numpy==1.26.4 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-api==1.15.0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-exporter-otlp-proto-grpc==1.15.0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-exporter-otlp-proto-http==1.15.0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-exporter-otlp==1.15.0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-instrumentation-grpc==0.36b0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-instrumentation==0.36b0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-proto==1.15.0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-sdk==1.15.0 ; python_version >= "3.9" and python_version < "3.13"
-opentelemetry-semantic-conventions==0.36b0 ; python_version >= "3.9" and python_version < "3.13"
+nvidia-cublas-cu12==12.1.3.1 ; platform_system == "Linux" and platform_machine == "x86_64" and python_version >= "3.9" and python_version < "3.13"
```

Comment:
why do we need all these nvidia packages?

Reply:
Those changes were added after I ran `make install && make update-lock && make export-requirements`

Reply:
We usually manually modify the requirements generated by Poetry to remove these dependencies and a few others. Note that there is now a dependency for `torch==2.4.0`, which we absolutely don't want to keep as this would erase HPU's torch.
So I think you can just remove all the new dependencies in this file and just keep the ones that are updated.